component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateUtil.java

```diff
@@ -138,8 +138,8 @@ public static boolean addCertificate(String username, X509Certificate x509Certif
      * @throws AuthenticationFailedException
      */
     public static boolean validateCertificate(String userName, AuthenticationContext authenticationContext,
-                                              byte[] certificateBytes, boolean isSelfRegistrationEnable)
-            throws AuthenticationFailedException {
+                                              byte[] certificateBytes, boolean isSelfRegistrationEnable,
+                                              boolean searchSecondaryUserStores) throws AuthenticationFailedException {
```

Comment:
We can't change the method signatures of the existing public methods . There can be external codes which rely on this public method of this Util. Hence we need to deprecate the existing method and introduce a new method with the new signature.

Reply:
I called the getX509Parameters() to get the parameter which is defined in the application-authentication.xml.
The "SearchAllUserStores" parameter should be defined as follows:
```<AuthenticatorConfig name="x509CertificateAuthenticator" enabled="true">
            <Parameter name="AuthenticationEndpoint">https://localhost:443/x509-certificate-servlet</Parameter>
            <!-- <Parameter name="username">CN</Parameter> -->
            <!-- <Parameter name="UsernameRegex">[a-zA-Z]{3}</Parameter> -->
            <Parameter name="AlternativeNamesRegex">[a-zA-Z]{7}</Parameter>
            <Parameter name="SearchAllUserStores">true</Parameter>
            <!-- <Parameter name="X509UserStoreName">SECONDARY</Parameter> -->
            <!--<Parameter name="setClaimURI">http://wso2.org/claims/identity/userCertificate</Parameter>-->
            <!--<Parameter name="EnforceSelfRegistration">true</Parameter>-->
        </AuthenticatorConfig>